### ISSUE TITLE
bug: GitHub Actions checking wrong commit

### ISSUE THREAD

**Issue by MikeSpreitzer** [kubestellar/kubestellar#809]
### Describe the bug

In https://github.com/kubestellar/kubestellar/actions/runs/5614367191 , which is a run of the docs-ecutable-example1.yml workflow for PR #713, the invocation of `actions/checkoug@v3.5.3` reported that it checked out commit f47734683fb824e2f92544de911b1f6c439d1d9d .  That was the head of the `main` branch at that time, not the commit to be tested for that PR.

### Steps To Reproduce

Look at the commit reported by `actions/checkoug@v3.5.3` and see if it is the commit that should be tested.

### Expected Behavior

The commit being tested for a PR should be a merge of the branch in the PR with the target branch.

**Comment by MikeSpreitzer** [comment]
Here is why that action fetched the `main` branch rather than the PR's merge commit:  https://github.com/kubestellar/kubestellar/blob/fa88851d7b43243362e00fe9d25fc179ab469ed9/.github/workflows/docs-ecutable-example1.yml#L8 uses event type `pull_request_target` rather than `pull_request`.  As documented at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target , this means that the workflow will be directed at the PR's base branch.  https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request says that for event type `pull_request`, on the other hand, the workflow would be directed to the PR's merge branch.

/assign @clubanderson 
@fileppb

**Comment by MikeSpreitzer** [comment]
https://github.com/kubestellar/kubestellar/blob/fa88851d7b43243362e00fe9d25fc179ab469ed9/.github/workflows/docs-ecutable-example1.yml#L2 also gives another clue.  https://github.com/kubestellar/kubestellar/actions/runs/5614367191 has "main" in its run-name where `${{ github.ref_name }}` is called for.

**Comment by clubanderson** [comment]
all docs-ecutable actions are now updated to use pull_request instead of pull_request_target.  Pull_request_target was used to protect against secret sharing.  Now that secrets are not needed in docs-ecutable, pull_request is sufficient again.

main, 0.2.0, 0.3.0, and 0.4.0 branches were all affected.

I also removed scheduler gh action for 0.4.0 and main since scheduler is replaced by where-resolver
